drivers/regulator/pmic_shell.c

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2022 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/zephyr.h>
+#include <zephyr/shell/shell.h>
+#include <zephyr/drivers/uart.h>
+#include <zephyr/drivers/regulator.h>
+#include <zephyr/drivers/regulator/consumer.h>
+
+LOG_MODULE_REGISTER(regulator_shell, CONFIG_REGULATOR_LOG_LEVEL);
+
+#define GET_REGULATOR(node) DEVICE_DT_GET(node),
+
+static const struct device *supported_pmics[] = {
+	DT_FOREACH_STATUS_OKAY(regulator_pmic, GET_REGULATOR)
+	};
+static const struct device *pmic_dev;
+
+#define MAX_REGULATOR_VOLTAGE 5000000 /* Max voltage of 5V */
```

Comment:
Why?

Reply:
This is legacy, since this shell was initially used to test the PCA9420, a PMIC regulator that won't go above that voltage. I've made the max voltage an argument to the shell command to set the PMIC voltage